src/rpcblockchain.cpp

```diff
@@ -813,6 +848,9 @@ bool AdvertiseBeacon(std::string &sOutPrivKey, std::string &sOutPubKey, std::str
             return false;
         }
 
+        // here try to import keys from config
+        ImportBeaconKeysFromConfig();
```

Comment:
If we move the import to the init we can remove this call and have advertise just advertise.